src/experimental/react/viewport-layout.js

```diff
@@ -0,0 +1,60 @@
+import React, {createElement, cloneElement} from 'react';
+
+export default class ViewportLayout extends React.Component {
+
+  constructor(props) {
+    super(props);
+  }
+
+  // Iterate over viewport descriptors and render viewports at specified positions
+  _renderViewportComponents() {
+    const {viewports} = this.props;
+
+    const children = [];
+
+    viewports.forEach((viewportOrDescriptor, i) => {
+      const {viewport, component, Component, props = {}} = viewportOrDescriptor;
+
+      // Check if this is a descriptor with a component, otherwise ignore
+      if (viewport && (component || Component)) {
```

Comment:
Nit : `component` and  `Component`, is confusing,  can we rename one of this.

Reply:
> Nit : component and Component, is confusing, can we rename one of this.

I can delete `Component` support for now.